### PR TITLE
JBPM-7338 Allow filtering manage section related items by container/d…

### PR DESCRIPTION
**Pull request by bxf12315**
…eployment on Process Instance List Page

### CODE REVIEW

**Comment by bxf12315** [comment]
@cristianonicolai It is first part of JBPM-7338, I will raise same code on task list page.

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/resources/org/jbpm/workbench/ht/client/resources/i18n/Constants.properties [hunk 1]
```diff
@@ -134,4 +134,5 @@ Released=Released
 Resumed=Resumed
 Delegated=Delegated
 Nominated=Nominated
+Container_Id= Container Id
```
Comment (cristianonicolai) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/resources/org/jbpm/workbench/ht/client/resources/i18n/Constants.properties]: remove extra space

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties [hunk 2]
```diff
@@ -179,4 +179,5 @@ TimerInstanceRescheduled=Timer instance {0} rescheduled
 FromNow=From now
 FromActivationTime=From activation time
 ShowInstanceBadges=Show instance badges
-TimerDelayHelp=In seconds
\ No newline at end of file
+TimerDelayHelp=In seconds
+Container_Id= Container Id
```
Comment (cristianonicolai) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties]: remove extra space

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java [hunk 3]
```diff
@@ -94,6 +94,8 @@ public void loadFilters() {
 
         addProcessNameFilter(getDataSetId());
 
+        addContainerIdFilter(getDataSetId());
```
Comment (cristianonicolai) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java]: @bxf12315 Container Id filter is rather technical, it shouldn't be added to the Task Inbox, only to Manage Tasks. Please add this logic to _TaskAdminListBasicFiltersPresenter_ only.

**Comment by cristianonicolai** [comment]
@bxf12315 please also add this new feature to the N&N section of 7.19 release

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties (hunk 2 above)
Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties]: ok

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/resources/org/jbpm/workbench/ht/client/resources/i18n/Constants.properties (hunk 1 above)
Comment (bxf12315) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/resources/org/jbpm/workbench/ht/client/resources/i18n/Constants.properties]: ok

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java (hunk 3 above)
Comment (bxf12315) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java]: just in admin, user can see the filter?

**Comment by bxf12315** [comment]
@cristianonicolai Updated. Please review it.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-api/src/main/java/org/jbpm/workbench/pr/model/ProcessInstanceDataSetConstants.java [hunk 4]
```diff
@@ -22,6 +22,7 @@
     public static final String PROCESS_INSTANCE_WITH_VARIABLES_DATASET = "jbpmProcessInstancesWithVariables";
     public static final String COLUMN_PROCESS_INSTANCE_ID = "processInstanceId";
     public static final String COLUMN_PROCESS_ID = "processId";
+    public static final String COLUMN_CONTAINER_ID = "externalId";
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-api/src/main/java/org/jbpm/workbench/pr/model/ProcessInstanceDataSetConstants.java]: got it.

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-api/src/main/java/org/jbpm/workbench/ht/model/TaskDataSetConstants.java [hunk 5]
```diff
@@ -22,6 +22,7 @@
     public static final String HUMAN_TASKS_WITH_ADMIN_DATASET = "jbpmHumanTasksWithAdmin";
     public static final String HUMAN_TASKS_WITH_VARIABLES_DATASET = "jbpmHumanTasksWithVariables";
 
+    public static final String COLUMN_CONTAINER_ID = "deploymentId";
```
[the remark this answers is not in the document]
Comment (bxf12315) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-api/src/main/java/org/jbpm/workbench/ht/model/TaskDataSetConstants.java]: got it.

**Comment by bxf12315** [comment]
updated, Please review again.

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/resources/org/jbpm/workbench/ht/client/resources/i18n/Constants.properties (hunk 1 above)
Comment (nmirasch) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/resources/org/jbpm/workbench/ht/client/resources/i18n/Constants.properties]: Regarding the process instance shows this info as 'Deployment' to be consistence with the process instance screen I  would rename this filter too.

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties (hunk 2 above)
Comment (nmirasch) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties]: I think the name at filters should point to 'Deployment'  instead 'Container Id' regarding that details are not displayed directly on any table column, but you can find them at Process Instance Details tab as 'Deployment'.

Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/resources/org/jbpm/workbench/pr/client/resources/i18n/Constants.properties]: I think you are right, it need align Process Instance Details

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java [hunk 6]
```diff
@@ -135,6 +135,20 @@ protected void addProcessNameFilter(final String dataSetId) {
                                                                   f.getValue())));
     }
 
+    protected void addContainerIdFilter(final String dataSetId) {
```
Comment (nmirasch) [inline jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java]: @bxf12315 would it be a better name for this function addDeploymentIdFilter regarding the filter is related to that COLUM_DEPLOYMENT_ID?

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java [hunk 7]
```diff
@@ -175,11 +177,25 @@ protected void addProcessIdFilter() {
                                     f -> addSearchFilter(f, equalsTo(COLUMN_PROCESS_ID, f.getValue())));
     }
 
+    protected void addContainerIdFilter() {
```
Comment (nmirasch) [inline jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: @bxf12315 here, the same, I think a better name is addDeploymentIdFilter, to align with the kind of information is related to.

File: jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java (hunk 6 above)
Comment (bxf12315) [reply jbpm-wb-human-tasks/jbpm-wb-human-tasks-client/src/main/java/org/jbpm/workbench/ht/client/editors/taskslist/AbstractTaskListBasicFiltersPresenter.java]: ok

File: jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java (hunk 7 above)
Comment (bxf12315) [reply jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListBasicFiltersPresenter.java]: ok

**Comment by bxf12315** [comment]
@cristianonicolai @nmirasch Updated Please review it.

**Comment by cristianonicolai** [comment]
@jstastny-cz happy for this to be merged?

**Pull request merged**